ledger/eval.go

```diff
@@ -1090,7 +1090,7 @@ func (eval *BlockEvaluator) finalValidation() error {
 		}
 	}
 
-	return nil
+	return eval.state.CalculateTotals()
```

Comment:
why here and not at the of `eval()` func?

Reply:
the eval function is good, but then we'll need to replicate the call to `eval.state.CalculateTotals` on `GenerateBlock() and `EvalForIndexer()`, which is error prone - I tried to make it more "structured".

Note of the changes in https://github.com/algorand/go-algorand/pull/2924 which also touches the same code path(s).